update-format-crau/src/delta_update.rs

```diff
@@ -88,21 +88,25 @@ pub fn get_signatures_bytes<'a>(mut f: &'a File, header: &'a DeltaUpdateFileHead
 #[rustfmt::skip]
 // parse_signature_data takes a bytes slice for signature and public key file path.
 // Return only actual data, without version and special fields.
-pub fn parse_signature_data(testdata: &[u8], sigbytes: &[u8], pubkeyfile: &str) -> Option<Box<[u8]>> {
+pub fn parse_signature_data(f: &mut BufReader<File>, sigbytes: &[u8], pubkeyfile: &str) -> Option<Box<[u8]>> {
     // Signatures has a container of the fields, i.e. version, data, and
     // special fields.
     let sigmessage = match proto::Signatures::parse_from_bytes(sigbytes) {
         Ok(data) => data,
         _ => return None,
     };
 
+    // Read the whole data to bufdata.
+    let mut testdata: Vec<u8> = Vec::new();
+    let _ = f.read_to_end(&mut testdata);
+
     // sigmessages.signatures[] has a single element in case of dev update payloads,
     // while it could have multiple elements in case of production update payloads.
     // For now we assume only dev update payloads are supported.
     // Return the first valid signature, iterate into the next slot if invalid.
     sigmessage.signatures.iter()
         .find_map(|sig|
-            verify_sig_pubkey(testdata, sig, pubkeyfile)
+            verify_sig_pubkey(testdata.as_slice(), sig, pubkeyfile)
             .map(Vec::into_boxed_slice))
 }
 
```

Comment:
Meta comment on the below function signature: What is the meaning of the return type? Could this be changed to Result with an error to be easier to understand and handle for the caller?

Reply:
As I remember, the first version of the code had a full Result type like you say. After a code review, we chose to drop that for simplicity.

Reply:
I don't understand what this returned array means, normally a verification either succeeds or gives an error.
@wrl Do you have a suggestion on how to improve this?